src/meta-srv/src/service/admin/maintenance.rs

```diff
@@ -91,13 +92,17 @@ impl MaintenanceHandler {
 impl HttpHandler for MaintenanceHandler {
     async fn handle(
         &self,
-        path: &str,
+        _: &str,
+        method: http::Method,
         params: &HashMap<String, String>,
     ) -> crate::Result<Response<String>> {
-        if path.ends_with("/set") {
-            self.set_maintenance(params).await
-        } else {
-            self.get_maintenance().await
+        match method {
+            http::Method::GET => self.get_maintenance().await,
+            http::Method::PUT => self.set_maintenance(params).await,
```

Comment:
In general, the PUT method sends data via the request payload. Are the params coming from the request payload?

Reply:
No. We're using query params still. Pass and decode body can be quite complicated. I tend not to implement it in this PR.